lib/editor/components/map/PatternStopsLayer.js

```diff
@@ -70,40 +76,77 @@ export default class PatternStopsLayer extends Component<Props> {
     if (cpIndex < patternStops.length) {
       console.warn(`Fewer control points (${controlPoints.length}) than pattern stops (${patternStops.length})!`, controlPoints, patternStops)
     }
+    const patternHalts = mergePatternHalts(patternStopsWithControlPointIndexes, patternLocations)
     return (
       <div id='PatternStops'>
-        {patternStopsWithControlPointIndexes.map((patternStop, index) => {
-          const {cpIndex, stopId} = patternStop
-          const stop = activePatternStops.find(s => s.stop_id === stopId)
-          if (!stop) {
-            console.log(stop)
-            console.warn(`Could not find stop for stopId: ${stopId}`, activePatternStops)
-            return
+        {patternHalts.map((patternStop, index) => {
+          if (patternStop.hasOwnProperty('stopId')) {
+            // $FlowFixMe Flow doesn't understand our type check
+            const { cpIndex, stopId } = patternStop
+            const stop = activePatternStops.find((s) => s.stop_id === stopId)
+            if (!stop) {
+              console.warn(
+                `Could not find stop for stopId: ${stopId}`,
+                activePatternStops
+              )
+              return
+            }
+            if (
+              editSettings.hideInactiveSegments &&
+              cpIndex && (cpIndex > patternSegment + 1 || cpIndex < patternSegment - 1)
+            ) {
+              // Do not render pattern stop if hiding inactive segments and
+              // pattern stop does not reference one of the adjacent control points.
+              return null
+            }
```

Comment:
This code should become a filter. (It's long, so probably extract a function for the filter)

Reply:
It is and is in need of refactoring but I'm weary of making changes to additional files. Don't want to introduce any more potential bugs